lib/widgets/editor/editor_toolbar.dart

```diff
@@ -0,0 +1,462 @@
+            before: '~~',
+            placeholder: L10n.of(context).insert_text_here_placeholder,
+          ),
+          icon: const Icon(Icons.format_strikethrough),
+          tooltip: L10n.of(context).editor_strikethrough,
+        ),
+        IconButton(
+          onPressed: () {
+            controller.insertAtBeginningOfEverySelectedLine('> ');
+          },
+          icon: const Icon(Icons.format_quote),
+          tooltip: L10n.of(context).editor_quote,
+        ),
+        IconButton(
+          onPressed: () {
+            final line = controller.firstSelectedLine;
+
+            if (line.startsWith(RegExp.escape('* '))) {
+              controller.removeAtBeginningOfEverySelectedLine('* ');
+            } else if (line.startsWith('- ')) {
+              controller.removeAtBeginningOfEverySelectedLine('- ');
+            } else {
+              controller.insertAtBeginningOfEverySelectedLine('- ');
+            }
+          },
+          icon: const Icon(Icons.format_list_bulleted),
+          tooltip: L10n.of(context).editor_list,
+        ),
+        IconButton(
+          onPressed: () => controller.surround(
+            before: '`',
+            placeholder: L10n.of(context).insert_text_here_placeholder,
+          ),
+          icon: const Icon(Icons.code),
+          tooltip: L10n.of(context).editor_code,
+        ),
+        IconButton(
+          onPressed: () => controller.surround(
+            before: '~',
+            placeholder: L10n.of(context).insert_text_here_placeholder,
+          ),
+          icon: const Icon(Icons.subscript),
+          tooltip: L10n.of(context).editor_subscript,
+        ),
+        IconButton(
+          onPressed: () => controller.surround(
+            before: '^',
+            placeholder: L10n.of(context).insert_text_here_placeholder,
+          ),
+          icon: const Icon(Icons.superscript),
+          tooltip: L10n.of(context).editor_superscript,
+        ),
+        //spoiler
+        IconButton(
+          onPressed: () {
+            controller.reformat((selection) {
+              final insides = selection.isNotEmpty ? selection : '___';
+              Logger.root
+                  .info([21, 21 + insides.length, insides, insides.length]);
+              return Reformat(
+                text: '\n::: spoiler spoiler\n$insides\n:::\n',
+                selectionBeginningShift: 21,
+                selectionEndingShift: 21 + insides.length - selection.length,
+              );
+            });
+          },
+          icon: const Icon(Icons.warning),
+          tooltip: L10n.of(context).editor_spoiler,
+        ),
+        IconButton(
+          onPressed: () {
+            launchLink(link: markdownGuide, context: context);
+          },
+          icon: const Icon(Icons.question_mark),
+          tooltip: L10n.of(context).editor_help,
+        ),
+      ],
+    );
+  }
+}
+
+class AddLinkDialog extends HookWidget {
+  final String title;
```

Comment:
This is called `text` or `label`, title is a different thing:
<img width="854" alt="image" src="https://user-images.githubusercontent.com/29288116/186010863-8bb81a02-c2d1-44e0-ae10-1ec21218fb10.png">